lib/facility.rb

```diff
@@ -1,14 +1,72 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, 
+              :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info)
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return unless @services.include?('Vehicle Registration')
+    return unless vehicle.is_a?(Vehicle)
+
+    update_plate_type_and_fees(vehicle)
+    set_registration_date(vehicle)
+    @registered_vehicles << vehicle
+  end
+
+  def administer_written_test(registrant)
+    return false unless @services.include?('Written Test')
+    return false unless registrant.permit? && registrant.age >=16
+
+    registrant.license_data[:written] = true
+    true
+  end
+  
+  def administer_road_test(registrant)
+    return false unless @services.include?('Road Test')
+    return false unless registrant.license_data[:written]
+    return false unless registrant.permit?
+
+    registrant.license_data[:license] = true
+    true
+  end
+
+  def renew_drivers_license(registrant)
+    return false unless @services.include?('Renew License')
+    return false unless registrant.license_data[:license]
+
+    registrant.license_data[:renewed] = true
+    true
+  end
+
+  private
+
+  def update_plate_type_and_fees(vehicle)
```

Comment:
This is good, however moving forward, we want to start thinking about Single Responsibility Principle (SRP) where every method should be doing one thing. The fact that you have an `and` in your method name is a great indicator that this should be split up into 2 methods.